ambry-tools/src/main/java/com.github.ambry/tools/admin/ServerAdminTool.java

```diff
@@ -245,11 +245,18 @@ public static void main(String args[]) throws Exception {
         }
         break;
       case TriggerCompaction:
-        ServerErrorCode errorCode = serverAdminTool.triggerCompaction(dataNodeId, config.partitionId, clusterMap);
-        if (errorCode == ServerErrorCode.No_Error) {
-          LOGGER.info("Compaction has been triggered for {} on {}", config.partitionId, dataNodeId);
+        if (config.partitionIds.length > 0 && !config.partitionIds[0].isEmpty()) {
+          for (String partitionId : config.partitionIds) {
+            ServerErrorCode errorCode = serverAdminTool.triggerCompaction(dataNodeId, partitionId, clusterMap);
+            if (errorCode == ServerErrorCode.No_Error) {
+              LOGGER.info("Compaction has been triggered for {} on {}", partitionId, dataNodeId);
```

Comment:
Do you think having an option whether to proceed on failure or throw an exception would help.